packages/wdio-utils/src/shim.ts

```diff
@@ -99,6 +146,121 @@ let wrapCommand = function wrapCommand<T>(commandName: string, fn: Function): (.
 
         return commandResult
     }
+
+    function wrapElementFn (promise: Promise<Clients.Browser>, cmd: Function, args: any[], prevInnerArgs?: { prop: string, args: any[] }): any {
+        return new Proxy(
+            Promise.resolve(promise).then((ctx: Clients.Browser) => cmd.call(ctx, ...args)),
+            {
+                get: (target, prop: string) => {
+                    /**
+                     * if we call a query method on a resolve promise, e.g.:
+                     * ```js
+                     * const elem = await $('foo')
+                     * await elem.$('bar')
```

Comment:
Can be simplified to the following right?
```suggestion
                     * await $('foo').$('bar')
```